### PR TITLE
pipelines: add lint-py to linter pipeline

### PR DESCRIPTION
**Pull request by refack**
fixes: #1631 

example output (`master`):
```
18:38:04  + python tools/getnodeversion.py
18:38:05  [Pipeline] sh
18:38:09  + NODE=node PYTHON=python3 gmake lint-py
18:38:09  PYTHONPATH=tools/pip python3 -m flake8 . \
18:38:09  	--count --show-source --statistics --select=E901,E999,F821,F822,F823 \
18:38:09  	--exclude=.git,deps,lib,src,test/fixtures,tools/*_macros.py,tools/gyp,tools/inspector_protocol,tools/jinja2,tools/markupsafe,tools/pip
18:38:18  0
18:38:19  [Pipeline] sh
18:38:22  + NODE=node gmake lint-ci
18:38:22  Running JS linter...
18:39:43  Running C++ linter...
18:40:19  PYTHONPATH=tools/pip python -m flake8 . \
18:40:19  	--count --show-source --statistics --select=E901,E999,F821,F822,F823 \
18:40:19  	--exclude=.git,deps,lib,src,test/fixtures,tools/*_macros.py,tools/gyp,tools/inspector_protocol,tools/jinja2,tools/markupsafe,tools/pip
18:40:25  0
```

`v10.x-staging`:
```
18:43:20  + python tools/getnodeversion.py
18:43:20  [Pipeline] sh
18:43:22  + NODE=node gmake lint-ci
```

`v8.x-staging` - https://ci.nodejs.org/job/node-linter/3040/console
```
19:19:36  + python tools/getnodeversion.py
19:19:37  [Pipeline] echo
19:19:37  8.15.2
19:19:37  [Pipeline] sh
19:19:38  + NODE=node gmake lint-ci
19:19:38  Running JS linter...
19:20:19  Running C++ linter on addon docs...
```

### CODE REVIEW

File: jenkins/pipelines/node-linter.jenkinsfile [hunk 1]
```diff
@@ -42,15 +42,22 @@ pipeline {
...
+                sh(script: "${env.MAKE} lint-py-build", returnStatus: true)
             }
         }
          
         stage('Run tests') {
             steps {
                 script {
+                    def node_version = sh(script: "@python  tools/getnodeversion.py", returnStdout: true)
+                    def lint_py3_ret = 0
+                    if (node_version > '11') {
```
Comment (richardlau) [inline jenkins/pipelines/node-linter.jenkinsfile]: Is this a safe comparison?

Comment (refack) [reply jenkins/pipelines/node-linter.jenkinsfile]: Tweaked just enough to get it to work with `8.`
```Groovy
node_version > "11." && node_version.take(1) != '8'
```

Comment (refack) [reply jenkins/pipelines/node-linter.jenkinsfile]: P.S. I don't think I like groovy

File: jenkins/pipelines/node-linter.jenkinsfile [hunk 2]
```diff
@@ -42,15 +42,23 @@ pipeline {
...
             }
         }
          
         stage('Run tests') {
             steps {
                 script {
+                    def node_version = sh(script: "python  tools/getnodeversion.py", returnStdout: true).trim()
+                    echo node_version
+                    def lint_py3_ret = 0
+                    if (node_version > "11." && node_version.take(1) != '8') {
```
Comment (richardlau) [inline jenkins/pipelines/node-linter.jenkinsfile]: I think doing a `>` comparison on a String is not a good idea. Suggest something like the following that splits the version string into a list and then converts the first element in the list to an int:
```groovy
def version_parts = node_version.tokenize('.')
def major = version_parts[0] as int
if (major > 11) {
...
```

**Comment by richardlau** [comment]
Oh and you've mispelled `linter` in the commit message and PR title 😆.

File: jenkins/pipelines/node-linter.jenkinsfile (hunk 2 above)
Comment (refack) [reply jenkins/pipelines/node-linter.jenkinsfile]: Ack

**Comment by refack** [comment]
New CI for master - https://ci.nodejs.org/job/node-linter/3048/console
New CI for v8.x - https://ci.nodejs.org/job/node-linter/3049/console

**Comment by refack** [comment]
v6.x Linter: https://ci.nodejs.org/job/node-linter/3116/ ✔️ 
v6.x Lite-CI: https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3146/ ✔️

**Comment by Trott** [comment]
/ping @nodejs/jenkins-admins to see if anyone else wants to take a look at this

**Comment by cclauss** [comment]
This is linking on both Python 2 and 3?

**Comment by richardlau** [comment]
> 
> 
> This is linting on both Python 2 and 3?

Yes for versions of Node.js > 11.

**Comment by sam-github** [comment]
@richardlau do you understand enough to land this, or call it stale and close it?

**Comment by richardlau** [comment]
These changes have already been made to the pipelines in question* so this PR is basically merging those changes to the backup jenkinsfile in this repository. We can have a separate discussion about whether having jenkinsfiles in SCM and not using those directly in the Jenkins jobs is a good idea or not but for now I'll just merge this in so the files here matches the scripts in the jobs.

* We haven't been running the `node-test-pull-request-lite-pipeline` job since we allowed just Travis CI for doc-only changes. I think there were reports over the Christmas period that it's possibly broken. Maybe we should just remove it? (Linting is done in `node-test-pull-request`/`node-test-commit` by a different lint job in Jenkins and we're also running the linters on Travis CI and GitHub Actions so it's not as if we're lacking coverage here.)

**Pull request merged**